include/onnxruntime/core/graph/constants.h

```diff
@@ -38,46 +38,51 @@ constexpr const char* kDmlExecutionProvider = "DmlExecutionProvider";
 constexpr const char* kMIGraphXExecutionProvider = "MIGraphXExecutionProvider";
 constexpr const char* kAclExecutionProvider = "ACLExecutionProvider";
 constexpr const char* kArmNNExecutionProvider = "ArmNNExecutionProvider";
-constexpr const char *providers_available[] = {
-  kCpuExecutionProvider,
+constexpr const char* providers_available[] = {
+    kCpuExecutionProvider,
 #ifdef USE_CUDA
-  kCudaExecutionProvider,
+    kCudaExecutionProvider,
 #endif
 #ifdef USE_DNNL
-  kDnnlExecutionProvider,
+    kDnnlExecutionProvider,
 #endif
 #ifdef USE_NGRAPH
-  kNGraphExecutionProvider,
+    kNGraphExecutionProvider,
 #endif
 #ifdef USE_OPENVINO
-  kOpenVINOExecutionProvider,
+    kOpenVINOExecutionProvider,
 #endif
 #ifdef USE_NUPHAR
-  kNupharExecutionProvider,
+    kNupharExecutionProvider,
 #endif
 #ifdef USE_VITISAI
-  kVitisAIExecutionProvider,
+    kVitisAIExecutionProvider,
 #endif
 #ifdef USE_TENSORRT
-  kTensorrtExecutionProvider,
+    kTensorrtExecutionProvider,
 #endif
 #ifdef USE_NNAPI
-  kNnapiExecutionProvider,
+    kNnapiExecutionProvider,
 #endif
 #ifdef USE_RKNPU
-  kRknpuExecutionProvider,
+    kRknpuExecutionProvider,
 #endif
 #ifdef USE_DML
-  kDmlExecutionProvider,
+    kDmlExecutionProvider,
 #endif
 #ifdef USE_MIGRAPHX
-  kMIGraphXExecutionProvider,
+    kMIGraphXExecutionProvider,
 #endif
 #ifdef USE_ACL
-  kAclExecutionProvider,
+    kAclExecutionProvider,
 #endif
 #ifdef USE_ARMNN
-  kArmNNExecutionProvider,
+    kArmNNExecutionProvider,
 #endif
 };
+
+// Key for disable MatMul PrePacking,
+// if the config value is set to "1" then the prepacking is disabled, otherwise prepacking is enabled
+constexpr const char* kDisablePrePacking = "Math.MatMul.DisablePrePacking";
```

Comment:
This is probably not the best place for these keys, any suggestion?

Reply:
I'd be tempted to have them under include/onnxruntime/core/common.

We should have some naming guidelines. I'd prefer an 'area' first - like 'op' for operators. Having Math and MatMul seems unnecessary (also I didn't think pre-packing would be limited to a specific operator, and that we would enable it more globally as that's what happens in SessionState currently).

---
In reply to: [469027840](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r469027840) [](ancestors = 469027840)